packages/dds/map/src/mapKernel.ts

```diff
@@ -628,43 +633,83 @@ export class MapKernel {
 				}
 				this.clearCore(local);
 			},
-			submit: (op: IMapClearOperation, localOpMetadata: IMapClearLocalOpMetadata) => {
+			resubmit: (op: IMapClearOperation, localOpMetadata: number) => {
+				const index = this.pendingMapLocalOpMetadata.findIndex(
```

Comment:
you could use a linked list here for better perf O(1) vs O(n), and less code. Specifically, make this.pendingMapLocalOpMetadata a linked list, and pass the node as the localOpMetadata. There is a list implementation in core-utils that is used for this sort of thing across a bunch of different dds

Reply:
Changed in latest